### ISSUE TITLE
Cater for multiple dropout rates for the various stages within Transformer model layers/heads

### ISSUE THREAD

**Issue by fuzing** [tracel-ai/burn#5129]
The current burn Transformer models use a single ``dropout`` rate for:
- multi-head attention
- cross attention (for decoder architecture)
- residual paths
- feed-forward

In many applications (e.g. a transformer encoder-only classification task containing infrequent label classes across training samples), a separate dropout rate would be used for each of these stages within a transformer encoder layer.

For example, in a transformer model handling infrequent (minority) labels, best practice would indicate the use of a moderate dropout rate of 0.2 to 0.4 across the attention weights and hidden layers, paired with an even lower dropout (0.05 to 0.1) on residual connections to avoid washing out the minority class signals.  For maximum flexibility, all 3/4 stages would be configurable.

**Note that this feature request is just a ``nice to have``**.  By way of comparison, Pytorch's transformer implementations also use a single dropout value across the various stages, necessitating the use of a bespoke/custom transformer setup to cater for this, which is identical to the current way burn handles this.

**Comment by nathanielsimard** [comment]
Transformers have a lot of options nowadays. I'm unsure of the right place to put such an implementation.
